### ISSUE TITLE
Performance degradation: convert -trim

### ISSUE THREAD

**Issue by fgagne** [ImageMagick/ImageMagick6#47]
I've upgraded from 6.9.10-29 to 6.9.10-43, version 32 bit for Windows.

Running `convert test-trim.png -trim test-trim-out.png` (with attached image) runs 2 times slower.

About 3 seconds vs 6 seconds run time between 6.9.10-29 and 6.9.10-43 (this is on a really old computer).

This seems like a  serious performance degradation.

![test-trim](https://user-images.githubusercontent.com/13245424/57106284-f20be380-6cfa-11e9-9d8c-6f2d134b1091.png)

**Comment by urban-warrior** [comment]
We implemented a more robust trimming algorithm that also supports removal of additional background, e.g. -fuzz 5% -define trim:percent-background=0% -trim.  We have an optimization in mind that should increase performance.  We'll need a few days to put together a patch.

**Comment by fgagne** [comment]
Thanks for working on this!

There is also a huge performance degradation between 6.9.10-29 and ImageMagick 7. Perhaps because of this new more robust trimming algorithm? Should I open an issue for ImageMagick 7?

**Comment by urban-warrior** [comment]
No need, we always track both distributions when we apply a patch.

**Comment by urban-warrior** [comment]
Thanks for the problem report.  We can reproduce it and will have a patch to fix it in GIT master branch @ https://github.com/ImageMagick/ImageMagick later today.  The patch will be available in the beta releases of ImageMagick @ https://www.imagemagick.org/download/beta/ by sometime tomorrow.

**Comment by fgagne** [comment]
I've tested ImageMagick-6.9.10-45_beta20190505 (1.29 sec) and the performance is in line with 6.9.10-29 (1.29 sec)! Thanks for the quick fix.

I've also tested ImageMagick-7.0.8-45_beta20190505 (2.45 sec) and it's an good improvement versus ImageMagick-7.0.8-43 (3.30 sec). (`magick convert test-trim.png -trim test-trim-out.png`)

However, there is still a huge performance gap between ImageMagick-6.9.10-45_beta20190505 (1.29 sec) and ImageMagick-7.0.8-45_beta20190505 (2.45 sec).

It doesn't seems to be related to `-trim`, see some example below (6.9.10-45_beta20190505 vs ImageMagick-7.0.8-45_beta20190505):
`convert test.png test_out.png` 1.36 sec VS 2.65 sec (PNG to PNG)
`convert test.png test_out.jpg` 0.43 sec VS 0.69 sec (PNG to **JPG**)
`convert test_out.jpg test_out_out.jpg` 0.30 sec VS 0.46 sec (**JPG** to **JPG**)

The timing above are on Windows, Q16, x86, Intel i5-2400.

I've also tested the x64 binaries, the performance gap is somewhat lower but still remains:
`convert test.png test_out.png` 1.26 sec VS 1.89 sec (PNG to PNG)
`convert test.png test_out.jpg` 0.41 sec VS 0.65 sec (PNG to **JPG**)
`convert test_out.jpg test_out_out.jpg` 0.30 sec VS 0.46 sec (**JPG** to **JPG**)

Is the performance gap between 6 and 7 specific to Windows?

**Comment by urban-warrior** [comment]
ImageMagick version 7, by default, has  [HDRI](https://imagemagick.org/script/high-dynamic-range.php) enabled.  HDRI requires float for pixel components.  Not only does this consume additional memory, float typically is slower in performance than integer processing.  For direct comparison, try the Q16, not the Q16HDRI release of IMv7.  How does the performance against IMv6 compare now?

```
magick --version
...
Features: ... HDRI ...
```

**Comment by fgagne** [comment]
First, I confirm the issue is fixed for ImageMagick version 6.9.10-48.

Thanks @urban-warrior for pointing out HDRI, I indeed version 7 tested with HDRI enabled (the portable version for Windows on the [download page](https://imagemagick.org/script/download.php#windows) has HDRI enabled even if it's not in the filename or the download description, ex: ImageMagick-7.0.8-48-portable-Q16-x86.zip). Would be nice if it would be made clearer.

I tested ImageMagick-6.9.10-48-Q16-x86 (1.27sec) VS ImageMagick-7.0.8-48-Q16-x86 **NO HDRI** (1.34sec). So everything looks fine :)

Thanks again for your help, closing the issue.
